OMEdit/OMEditLIB/Animation/AbstractVisualizer.h

```diff
@@ -95,11 +95,53 @@ class VisualizerAttribute
   unsigned int fmuValueRef;
 };
 
+template<typename type>
+class VisualProperty
+{
+public:
+  VisualProperty() {reset();}
+  virtual ~VisualProperty() = default;
+  virtual bool custom() const {return mCustomProperty;}
+  virtual type get(const type& rProperty = kDefaultProperty) const {return mCustomProperty ? mProperty : rProperty;}
+  virtual void set(const type& rProperty) {mCustomProperty = true, mProperty = rProperty;}
+  virtual void reset() {mCustomProperty = false, mProperty = kDefaultProperty;}
+private:
+  static const type kDefaultProperty;
+  bool mCustomProperty;
+  type mProperty;
+};
+
+class VisualProperties
+{
+public:
+  using type1 = QColor;
+  using type2 = float;
+  using type3 = std::string;
```

Comment:
Is there any point in these now? It seems like it would be better to just use the types directly instead of having aliases for them.